GSoC-2021/Kush-Kothari.md

```diff
@@ -0,0 +1,82 @@
+  - Recorded voice (Just like the Crisis Logger website mentioned in the list)
+  - Recorded video using connected webcam.
+  - Optional upload of image, sound and video files 
+    - The site will allow the user to upload a file.
+    - Resulting entire form will be authenticated in the backend via Django forms.
+    - Thus the form will be considered valid only if the user uploaded the correct file type (.jpg, .png, .mp4, .mp3 ...etc.) to the form.
+- Once the user scrolls down they will also get to select who has the data access permissions. This will be another field in the Model for UserExperience
+- Details(Fields) of a Model of UserExperience will be
+  - Primary Key (as generated by Django)
+  - User (a relation to the Users database which will be instances of users as formed by the [OpenHumansMemberObject](https://django-open-humans.readthedocs.io/en/latest/modules/interactions.html#accessing-an-openhumansmember-object) from the [django-open-humans](https://django-open-humans.readthedocs.io/en/latest/index.html) library)
+  - Type: A kind of an enum (enforced via validation rules) which will store the type of experience it is from the 5 mentioned above.
+  - Content (to store the actual content or reference to a file)
+  - Permissions: To store the kind of permissions the user has given for viewers and researchers.
+  - Associated Tags: For filtering options when searching up other experiences.
+  - Date: Date on which the experience was published
+  - Time: So that we can show details like "published 20mins ago" on the view other experiences page.
+  - Meta Data: Other possible meta data that could be associated with each post.
+
+The form filling flow will be as follows:
+- User is greeted with the appropriate title, text and the options:
+  - Text
+  - Image
+  - Sound
+  - Video
+- Once the user clicks on these further suboptions will appear depending on the choice selected. The sub-options are mapped as:
+  - Text
+    - Simple text
+    - Markdown
+  - Image
+    - Upload image
+    - Take image from Webcam
+  - Sound
+    - Record sound using microphone
+    - Upload a sound file
+  - Video
+    - Record video via webcam and microphone
+    - Upload a video file
+- The user gets to now change permissions and add comma-seperated tags for the experience
+- There will be a checkbox for adult content (the form will automatically add the "adult content" tag to the tag list when submitting this to the backend.)
+- A consent radio button
+- On selecting "Submit" the user will either be redirected to a confirmation page (in case of no error in the form validation) or the same page but now with a red error popup (in case of a faliure in the backend).
+- The confirmation page will now have easy navigation buttons just like its mentioned in the mockups on Figma. Options include:
+  - Enter another Experience
+  - Go to the My Experiences page
+  - View other Experiences
+  - Find out more about AutSPACEs
```

Comment:
Thank you @kkothari2001 - this looks really thorough.

Reply:
Thank you! I tried to make it as detailed as possible. Any more suggestions regarding the same?